projects/canopy/src/lib/carousel/carousel.component.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  AfterContentInit,
+  Component,
+  ContentChildren,
+  ElementRef,
+  Input,
+  QueryList,
+} from '@angular/core';
+
+import { HeadingLevel } from '../heading';
+import { LgCarouselItemComponent } from './carousel-item/carousel-item.component';
+
+@Component({
+  selector: 'lg-carousel',
+  templateUrl: './carousel.component.html',
+  styleUrls: ['./carousel.component.scss'],
+})
+export class LgCarouselComponent implements AfterContentInit {
+  @Input() description: string;
+  @Input() headingLevel: HeadingLevel;
+  @Input() loopMode = false;
+  @Input() slideDuration = 500;
+
+  @ContentChildren(LgCarouselItemComponent, { read: ElementRef })
+  carouselItems = new QueryList<ElementRef>();
+  selectedElementRef: ElementRef;
```

Comment:
does this need to be public?

Reply:
it breaks the ngFor in the  template if set to private